### PR TITLE
change Forc.toml to take an array of authors

### PR DESCRIPTION
**Pull request by JoshuaBatty**
Currently in the `Forc.toml` we only specify a single `author` as opposed to the `Cargo.toml` that allows for an array of `authors`. 

This PR changes `Forc.toml` to take an array of `authors` so it's structured the same as `Cargo.toml` files.

### CODE REVIEW

**Comment by JoshuaBatty** [comment]
Currently, CI is failing to build `sway-lib-core` as the manifest file is still only specifying a single author instead of an array. I've opened a PR up for that [here](https://github.com/FuelLabs/sway-lib-core/pull/11) 

Once this PR lands I'll need to go through all of the repos to submit PR's for updating the `Forc.toml` files.

**Comment by JoshuaBatty** [comment]
I've opened up a bunch of PR's below so that when this PR lands, the other repo's will be compatible with the change.

`sway-lib-core` = https://github.com/FuelLabs/sway-lib-core/pull/11
`sway-lib-std` = https://github.com/FuelLabs/sway-lib-std/pull/56
`fuels-rs` = https://github.com/FuelLabs/fuels-rs/pull/109
`fuels-ts` = https://github.com/FuelLabs/fuels-ts/pull/135
`swayswap-demo` = https://github.com/FuelLabs/swayswap-demo/pull/11

**Comment by adlerjohn** [comment]
I think the std and core libs go first, then this, then cut a new release, then the downstream?

**Comment by JoshuaBatty** [comment]
> I think the std and core libs go first, then this, then cut a new release, then the downstream?

Yeah, that plan sounds good. I'll wait until this PR gets approval from @digorithm first before merging the approved PR's in `sway-lib-core` and `sway-lib-std`

**Comment by adlerjohn** [comment]
Suggestion to make our life easier: instead introduce a _new_ field `authors` and keep the current `author` field. Then we can cut a release and migrate things over slowly without breaking everything with a circular dependency.

**Comment by JoshuaBatty** [comment]
Hi @adlerjohn I've made the `author` field take `Option<String>` and `authors` field take `Option<Vec<String>>` so we can slowly move away from using `author` without disrupting other projects. I've also added a deprecation notice on `author`. 

I think this should be good to go now.

**Review by adlerjohn — APPROVED** [review]
That was much easier than the first attempt 😂

**Pull request merged**
